### ISSUE TITLE
undefined method `ref' for nil:NilClass

### ISSUE THREAD

**Issue by gms8994** [voxpupuli/puppet-nodejs#185]
Running `puppet agent --test` against a puppetmaster with puppetlabs-nodejs v0.8.0 results in the error mentioned. 0.7.1 does not have this problem. Puppet 3.4.3 on master and agent. The following modules installed.

```
puppet module list
/etc/puppet/modules
├── maestrodev-ssh_keygen (v1.3.1)
├── puppetlabs-apt (v2.2.0)
├── puppetlabs-concat (v1.2.5)
├── puppetlabs-firewall (v1.7.1)
├── puppetlabs-nodejs (v0.8.0)
├── puppetlabs-stdlib (v4.9.0)
└── saz-ssh (v2.8.1)
```

```
puppet agent --test
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: undefined method `ref' for nil:NilClass on node `nodename`
```

**Comment by juniorsysadmin** [comment]
@gms8994 As discussed in #149 this appears to be due to a nested ```contain``` bug that was fixed in Puppet 3.7.0

**Comment by igalic** [comment]
https://github.com/voxpupuli/puppet-nodejs/commit/85b9b729bc6874660b6ad5e8487f23eec1cc8f1e
